Source/CesiumRuntime/Public/CesiumGltfPointsSceneProxyUpdater.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include "CesiumPointCloudShading.h"
+#include "Templates/SharedPointer.h"
+#include <glm/vec3.hpp>
+
+class ACesium3DTileset;
+class UCesiumGltfPointsComponent;
+class FCesiumGltfPointsSceneProxy;
+
+/**
+ * Used to pass tile data and Cesium3DTileset settings to a render thread to
+ * update the SceneProxy.
+ */
+struct FCesiumGltfPointsSceneProxyTilesetData {
+  FCesiumPointCloudShading PointCloudShading;
+  double MaximumScreenSpaceError;
+  bool UsesAdditiveRefinement;
+  float GeometricError;
+  glm::vec3 Dimensions;
+
+  FCesiumGltfPointsSceneProxyTilesetData();
+
+  void UpdateFromComponent(UCesiumGltfPointsComponent* Component);
+};
+
+/**
+ * This is used by a Cesium3DTileset to propagate its settings to any glTF
+ * points components it parents.
+ */
+class FCesiumGltfPointsSceneProxyUpdater {
+public:
+  FCesiumGltfPointsSceneProxyUpdater(ACesium3DTileset* Tileset);
+  ~FCesiumGltfPointsSceneProxyUpdater();
+
+  /** Updates proxies with new tileset settings. Must be called from a game
+   * thread. */
+  void UpdateSettingsInProxies();
+
+  /** Updates only if marked as needing an update. */
+  void UpdateSettingsInProxiesIfNeeded();
+
+  /** Marks this as needing an update next frame. This is called from
+   * CreateSceneProxy because the proxy won't be assigned to the primitive
+   * component until after that function. */
+  void MarkForUpdateNextFrame();
+
+private:
+  ACesium3DTileset* Owner;
+  bool bNeedsUpdate;
```

Comment:
I think you could get rid of this `bNeedsUpdate` field just by updating a new scene proxy immediately after it's created. Scene proxy are created in the game thread, so there shouldn't be any drama.

And then once that field is gone, this class doesn't need to hold any state at all anymore, which means it could just be static methods that take a ACesium3DTileset pointer and a tileset wouldn't need to hold an instance of it.